src/components/LeftPane/Drawers/ServiceAreasDrawer/ServiceAreasDrawer.tsx

```diff
@@ -60,32 +65,91 @@ function getCounties(serviceAreas: [string, string][]): string[] {
 }
 
 const COLUMNS = [
-  { aliases: ['CountyName'], required: true },
-  { aliases: ['ZipCode'], required: true }
+  { aliases: ['CountyName', 'county'] },
+  { aliases: ['ZipCode', 'zip'] }
 ]
 
-let parse = parseRows(COLUMNS, (([county, zip]) => {
-  let _county = capitalizeWords(county!)
+let parse = parseRows(COLUMNS, (([county, zip]): [string, string][] => {
 
-  // validate that county exists
-  if (!(_county in COUNTIES_TO_ZIPS)) {
-    throw `County "${county}" is not supported`
-  }
+  // We infer missing zips and counties, so one zip might map to
+  // more than one county, and one county maps to many zips.
+  let pairs = getCountyAndZip(county, zip)
+    .map(([county, zip]) => [capitalizeWords(county), zip] as [string, string])
+
+  // validate that counties exist
+  pairs.forEach(([county]) => {
+    if (!(county in COUNTIES_TO_ZIPS)) {
+      throw `County "${county}" is not supported`
+    }
+  })
 
   // validate that zip code is in county
   // TODO: consider pre-hashing zips for O(1) lookup
-  if (!COUNTIES_TO_ZIPS[_county].includes(zip!)) {
-    throw `Zip ${zip} does not exist in county "${county}"`
+  pairs.forEach(([county, zip]) => {
+    if (!COUNTIES_TO_ZIPS[county].includes(zip)) {
+      throw `Zip ${zip} does not exist in county "${county}"`
+    }
+  })
+
+  return pairs
+    .uniqBy(spread(serviceArea))
+    .value()
+}), validateHeaders)
+
+function validateHeaders(columns: ColumnDefinition[], fields: string[]) {
+  if (isEmpty(fields[0]) && isEmpty(fields[1])) {
+    return [new ParseError(0, 0, columns[0], fields, `CSV must define columns "CountyName" and/or "ZipCode"`)]
+  }
+  return []
+}
+
+function getCountyAndZip(county: string | null, zip: string | null): Lazy<[string, string][]> {
+  switch (getParseMode(county, zip)) {
+
+    case ParseMode.INFER_COUNTY:
+      // If county isn't defined but zip is, default to all counties
+      // that contain the given zip.
+      return chain(countiesFromZip(zip!))
+        .map(county => [county, zip] as [string, string])
+
+    case ParseMode.INFER_ZIP:
+      // If county is defined but zip isn't, default to all zips for
+      // the given county.
+      return chain(zipsFromCounty(capitalizeWords(county!)))
+        .map(zip => [county, zip] as [string, string])
+
+    case ParseMode.WELL_FORMED:
+      return chain([[county, zip] as [string, string]])
+
+    case ParseMode.INSUFFICIENT_DATA:
```

Comment:
Are we raising an error in this case?

Reply:
We don't raise errors! We pass a description of the error up to the consumer, where we can expose it to the user. See above https://github.com/bayesimpact/tds-frontend/pull/48/files#diff-b37cea03ced710e0e8dc6ac44dd784efR101